tests/osgi-tests/src/test/java/org/wso2/carbon/osgi/transport/TransportManagerTest.java

```
@@ -0,0 +1,127 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.osgi.transport;
+
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.ops4j.pax.exam.testng.listener.PaxExam;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+import org.wso2.carbon.kernel.transports.TransportManager;
+
+import javax.inject.Inject;
+
+/**
+ * OSGi tests class to test org.wso2.carbon.kernel.transports.TransportManager.
+ */
+@Listeners(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class TransportManagerTest {
+    private static final Logger logger = LoggerFactory.getLogger(TransportManagerTest.class);
+
+    @Inject
+    private TransportManager transportManager;
+
+    @Test
+    public void testTransportManagerExistence() {
+        Assert.assertNotNull(transportManager, "TransportManager Service is null");
+    }
+
+    @Test(dependsOnMethods = {"testTransportManagerExistence"})
+    public void testUnsuccessfulStartTransport() {
+        try {
+            transportManager.startTransport("wrongId");
+        } catch (IllegalArgumentException e) {
+            String exceptionMessage = "wrongId not found";
+            Assert.assertEquals(exceptionMessage, e.getMessage());
+        }
+    }
+
+    @Test(dependsOnMethods = {"testUnsuccessfulStartTransport"})
+    public void testSuccessfulStartTransport() {
+        try {
+            CustomCarbonTransport carbonTransport = new CustomCarbonTransport("dummyTransport");
+            transportManager.registerTransport(carbonTransport);
+            transportManager.startTransport("dummyTransport");
+        } catch (IllegalArgumentException e) {
```

Let's throw this error, so it'll be handled by the test runtime.